### PR TITLE
feat(connector): [Square] Use connector_response_reference_id as reference to merchant

### PR DESCRIPTION
**Pull request by hegdeadithyak**
## Type of Change

- [ ] Bugfix
- [x] New feature
- [ ] Enhancement
- [ ] Refactoring
- [ ] Dependency updates
- [ ] Documentation
- [ ] CI/CD

### Additional Changes

- [ ] This PR modifies the API contract
- [ ] This PR modifies the database schema
- [ ] This PR modifies application configuration/environment variables

## Checklist

- [ ] I formatted the code `cargo +nightly fmt --all`
- [ ] I addressed lints thrown by `cargo clippy`
- [ ] I reviewed the submitted code
- [ ] I added unit tests for my changes where possible
- [ ] I added a [CHANGELOG](/CHANGELOG.md) entry if applicable

### CODE REVIEW

**Comment by Adithya4720** [comment]
@deepanshu-iiitu  Can you check this out

**Comment by swangi-kumari** [comment]
Hey @Adithya4720 ,
Can you please resolve conflicts.

**Comment by Adithya4720** [comment]
@swangi-kumari Please Check this out

File: crates/router/src/connector/powertranz.rs [hunk 1]
```diff
@@ -87,6 +87,10 @@ impl ConnectorCommon for Powertranz {
         "application/json"
     }
 
+    fn get_currency_unit(&self) -> api::CurrencyUnit {
```
Comment (srujanchikke) [inline crates/router/src/connector/powertranz.rs:90]: Hi @Adithya4720 ,

This changes are not concerned with issue attached. Please refer #2052 for more info and also change the PR title accordingly.

**Review by srujanchikke — CHANGES_REQUESTED** [review]

**Pull request closed without merging**